scripts/removeUnusedResourcesFromAssets.swift

```diff
@@ -0,0 +1,104 @@
+import Foundation
+
+if CommandLine.arguments.count != 3 {
+    print("usage: swift removeUnusedResourcesFromAssets.swift <xcassets_file_path> <project_root_path>")
+} else {
+    let xcassetsPath = CommandLine.arguments[1]
+    let rootPath = CommandLine.arguments[2]
+
+    let usedResources = findUsedResources(in: rootPath)
+
+    removeResources(from: xcassetsPath,
+                    notContainedIn: usedResources)
+}
+
+/// Builds a set of resource entries relative to the root of an .xcassets folder based on contents of the .xcfilelist files in the project.
+/// - Parameter rootPath: Root path of the project. A search for .resources.xcfilelist files will be performed to build the resource entry list.
+/// - Returns: A set containing the combination of all resource entries in the .resources.xcfilelist found in the project.
+func findUsedResources(in rootPath: String) -> Set<String> {
+    let rootURL = URL(fileURLWithPath: rootPath)
+    let resourceFileSuffix = ".resources.xcfilelist"
+    var usedResources: Set<String> = []
+
+    print("Parsing *\(resourceFileSuffix) files in path \(rootURL)")
+    if let filesEnumerator = FileManager.default.enumerator(at: rootURL,
+                                                            includingPropertiesForKeys: [.isRegularFileKey],
+                                                            options: [.skipsHiddenFiles, .skipsPackageDescendants]) {
+        for case let fileURL as URL in filesEnumerator {
+            do {
+                let fileAttributes = try fileURL.resourceValues(forKeys: [.isRegularFileKey])
+                let filePath = fileURL.relativePath
+
+                if fileAttributes.isRegularFile! &&
+                    filePath.hasSuffix(resourceFileSuffix) {
+
+                    print("\nUsed resources in file: \(filePath)")
+
+                    do {
+                        let resourceFileListContents = try String(contentsOf: fileURL)
+
+                        for entry in resourceFileListContents.split(separator: "\n") {
+                            let resourceFileEntry = entry.trimmingCharacters(in: .whitespacesAndNewlines)
+                            usedResources.insert(resourceFileEntry)
+                            print("- \(resourceFileEntry)")
+                        }
+                    } catch {
+                        fatalError("Failed to read contents resource file: \(filePath) \nError: \(error)")
```

Comment:
Will the script fail if we reach this scenario?

Reply:
Although this is not expected to happen, the script will fail if we reach this scenario which will cause the build to fail.
The developer of the client app can inspect the log ([convo about whether it should be logged all the time](https://github.com/microsoft/fluentui-apple/pull/492#discussion_r602471442)) or directly inspect the project folder (FluentUI-ios.xcassets) or even inspect the resource bundle that was generated if the build succeeded.